src/main/java/studio/magemonkey/fusion/cfg/Cfg.java

```diff
@@ -1,182 +1,125 @@
-//                        new RecipeCustomItem(new ItemStack(Material.COOKIE), 2, true)),
-//                new RecipeEconomyItem("resultItem", 4), 0, 0, 0));
-        CraftingTable b = new CraftingTable("craft",
-                "Craft inventory name",
-                new VanillaProvider.VanillaItemType(Material.PAPER),
-                ip,
-                item/*new ItemStack(Material.BLACK_STAINED_GLASS_PANE)*/,
-                0,
-                0);
-        List<Map<String, Object>> list = new ArrayList<>(3);
-//        list.add(a.serialize());
-        list.add(b.serialize());
-        cfg.addDefault("types", list);
+        // addDefault did not add it to config, so here is a work around
+        if(!cfg.isSet("recursive_level_formula")) cfg.set("recursive_level_formula", recursive);
+        if(!cfg.isSet("final_level_mod")) cfg.set("final_level_mod", dataSaveInterval);
+        if(!cfg.isSet("data_save_interval")) cfg.set("data_save_interval", dataSaveInterval);
+        if(!cfg.isSet("forget.penalty")) cfg.set("forget.penalty", forgetPenalty);
+        if(!cfg.isSet("crafting_queue")) cfg.set("crafting_queue", craftingQueue);
+        if(!cfg.isSet("finished_message_interval")) cfg.set("finished_message_interval", 300);
     }
 
     public static void init() {
-        map.clear();
-        guiMap.clear();
-        File              file = new File(Fusion.getInstance().getDataFolder(), "config.yml");
-        FileConfiguration cfg  = new YamlConfiguration();
+        File file = new File(Fusion.getInstance().getDataFolder(), "config.yml");
+        FileConfiguration cfg = new YamlConfiguration();
 
         if (!file.exists()) {
             addDefs(cfg);
             file.getAbsoluteFile().getParentFile().mkdirs();
             try {
                 file.createNewFile();
             } catch (IOException e) {
-                Fusion.getInstance().getLogger().warning("Can't create config file: " + file);
-                e.printStackTrace();
+                Fusion.getInstance().getLogger().warning("Can't create config file: " + file + ":" + e.getMessage());
             }
             cfg.options().copyDefaults(true);
             try {
                 cfg.save(file);
             } catch (IOException e) {
-                Fusion.getInstance().getLogger().warning("Can't save config file: " + file);
-                e.printStackTrace();
+                Fusion.getInstance().getLogger().warning("Can't save config file: " + file + ":" + e.getMessage());
             }
         } else {
             try {
-                cfg.load(file);
+                reload(cfg, file);
             } catch (Exception e) {
-                Fusion.getInstance().getLogger().warning("Can't load config file: " + file);
-                e.printStackTrace();
+                Fusion.getInstance().getLogger().warning("Can't load config file: " + file + ":" + e.getMessage());
                 return;
             }
-            addDefs(cfg);
```

Comment:
Any reason to remove this call here?

Reply:
The addDefs method doesnt have worked on the init method because it was called after the file was saved. So I decided to make a method `reload` that automatically puts addDefs into the right place where it needs to be in order to save properly.